api/autopilot.go

```diff
@@ -4,38 +4,35 @@ import (
 	"errors"
 	"fmt"
 
-	"go.sia.tech/core/types"
 	"go.sia.tech/renterd/internal/utils"
 )
 
 const (
 	// BlocksPerDay defines the amount of blocks that are mined in a day (one
 	// block every 10 minutes roughly)
 	BlocksPerDay = 144
-
-	// DefaultAutopilotID is the id of the autopilot.
-	DefaultAutopilotID = "autopilot"
 )
 
 var (
-	// ErrAutopilotNotFound is returned when an autopilot can't be found.
-	ErrAutopilotNotFound = errors.New("couldn't find autopilot")
-
-	// ErrMaxDowntimeHoursTooHigh is returned if the autopilot config is updated
+	// ErrMaxDowntimeHoursTooHigh is returned if the contracts config is updated
 	// with a value that exceeds the maximum of 99 years.
 	ErrMaxDowntimeHoursTooHigh = errors.New("MaxDowntimeHours is too high, exceeds max value of 99 years")
+
+	// ErrInvalidReleaseVersion is returned if the version is an invalid release
+	// string.
+	ErrInvalidReleaseVersion = errors.New("invalid release version")
 )
 
 type (
-	// Autopilot contains the autopilot's config and current period.
+	// Autopilot contains its configuration as well as the current period.
 	Autopilot struct {
-		ID            string          `json:"id"`
-		Config        AutopilotConfig `json:"config"`
-		CurrentPeriod uint64          `json:"currentPeriod"`
+		CurrentPeriod uint64 `json:"currentPeriod"`
+		AutopilotConfig
 	}
 
-	// AutopilotConfig contains all autopilot configuration.
+	// AutopilotConfig contains configuration settings for the autopilot.
 	AutopilotConfig struct {
+		Enabled   bool            `json:"enabled"`
```

Comment:
Should be in the config file. Changing this on the fly is unnecessary.

Reply:
To be fair I can also not think of many reasons why you would want to flip it off and on _with_ a restart except for when you want to stop migrations. But it depends on who the user you are thinking of is. The one who is eventually going to use the lightweight client anyway or the one running the indexing service who might want to disable migrations without downtime for their users?

Reply:
Why would an indexing service want to disable migrations temporarily?

Reply:
Under perfect conditions you probably wouldn't. But if something goes wrong it can be useful. e.g. you have a ton of host churn / migrations for reasons that are just out of your control. A few things of which we experienced some could be:
- DNS resolvers acting up and your hosts seem to have gone bad
- Laptop standby causing the autopilot to run but internet to be offline leading to churn
- Temporarily running out of disk space or other db related issues causing issues
- DDoS attack

Reply:
> DNS resolvers acting up and your hosts seem to have gone bad

Resolving this is likely to involve a server/container restart anyway

> Laptop standby

- Is `renterd` designed to be run on laptops that go to sleep? There are a lot of processes that are expected to run on time outside of the autopilot. 
- Having to remember to turn off a setting before closing a laptop lid is terrible UX. 
- This user group is more likely to stop `renterd` entirely.

> Temporarily running out of disk space or other db related issues causing issues

Will likely need to restart either the database or the server to resolve. DB transactions outside of the autopilot will also be failing and creating havoc.

> DDoS attack

I can agree with this one. However, the likelihood of users being able to access the `renterd` service during a DDoS is unlikely to be affected by whether autopilot is on or off.

Reply:
I'm not trying to say these are the exact 4 issues we have this flag for. Just brought these up because usually we can't predict every single scenario and keeping this behavior doesn't cost us anything. 

A more practical, non-emergency scenario might be running a staging server with an updated version in parallel to a production server and then swapping them by disabling the autopilot on the old one and enabling it on the updated one for updates without downtime after updating your load balancer to redirect to the updated server for a bit. 

If you are sure there is no scenario where it might be useful then we can get rid of it. I just don't have that confidence.